### ISSUE TITLE
scalar: report error when we need to move decimal's zero point

### ISSUE THREAD

**Issue by skyzh** [singularity-data/risingwave#3971]
```
dev=> select 10000000000000000000000000000 + 0.0000000000000000001::decimal;
           ?column?
-------------------------------
 10000000000000000000000000000
```

which means that we lost precision while computing decimal. This should not be allowed, and we should report an error.

cc @fuyufjh

**Comment by fuyufjh** [comment]
Hi, any updates?

**Comment by nanderstabel** [comment]
> Hi, any updates?

Not yet, finding a solution for properly implementing BigDecimal turned out to be way more complex than I initially thought so I have paused this for now. However, I can Implement the report error and leave the implementation of BigDecimal to support larger precision for a later moment?

**Comment by xxchan** [comment]
> Not yet, finding a solution for properly implementing BigDecimal turned out to be way more complex than I initially thought so I have paused this for now. However, I can Implement the report error and leave the implementation of BigDecimal to support larger precision for a later moment?

Yes, IIUC a more powerful decimal is not necessary at least for now, since it's too complicated as discussed. But we should at least be correct, and reject errors e.g. in this issue 😄
